### PR TITLE
Adding ISO-8601 Date tests for functionality that was not covered.

### PR DESCRIPTION
**Pull request by gregoryosgood**
**Things To Remember:** 

1. Be sure to merge into the right branch. The choices are:
* **`master`** if your PR runs on the current version of Tableau.
* **`dev-2021.2`**  
2. You should update at least one of the following:
* The changelog of the project you modified.
* The root `README.md` if the PR is a release of the SDK, Connector Packager, or TDVT.  
=====

### CODE REVIEW

**Comment by rosswbrown** [comment]
Tracked in 1419930

File: tdvt/tdvt/exprtests/standard/setup.cast.str.datetime.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+DATETIME('1900-01-01 01:00:00')
+DATETIME('1900-12-30 12:00:00')
+DATETIME('1900-12-31 23:59:59')
+DATETIME('2018-01-01 01:00:00')
+DATETIME('2018-12-29 12:00:00')
+DATETIME('2018-12-31 23:59:59')
+
```
Comment (lukewrites) [inline tdvt/tdvt/exprtests/standard/setup.cast.str.datetime.txt:7]:
```suggestion

```

File: tdvt/tdvt/exprtests/standard/setup.cast.str.datetime.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+DATETIME(STR([datetime0]))
+DATETIME(STR([datetime1]))
```
Comment (lukewrites) [inline tdvt/tdvt/exprtests/standard/setup.cast.str.datetime.txt:9]: This test case is failing on a number of ds that cannot handle the cast on NULL or empty string. Let's split it off into a new test file maybe? (`setup.cast.str.datetime.empty`)

File: tdvt/tdvt/exprtests/standard/expected.setup.cast.str.datetime.txt [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+      </tuple>
+    </table>
+  </test>
+  <test name='DATETIME(STR([datetime0]))'>
+    <table>
+      <schema>
+        <column>[cast_calcs.sqlserver].[TEMP(Test)(3072781275)(0)]</column>
+      </schema>
+      <tuple>
+        <value>#2004-07-04 22:49:28#</value>
```
Comment (lukewrites) [inline tdvt/tdvt/exprtests/standard/expected.setup.cast.str.datetime.txt:68]: Hi @gregoryosgood when I run these tests against sqlserver now this test fails - sqlserver converts the seconds value to `00`. Reproduces across platforms. Wondering how this was generated.

**Pull request closed without merging**